无法缓存 SP

### 问题描述

缓存 SP 时无反应。

### 复现步骤

尝试手动缓存 Silent Witch 沉默魔女的秘密 的 SP9。

### Ani 版本号

5.0.0-beta02

### 操作系统

Windows

### 应用日志

```log
2025-09-12 14:14:46,843 [INFO ] app.torrent.anitorrent.AnitorrentTorrentDownloader: HTTP torrent file 'https://mikanani.me/Download/20250906/de81f8d99458eea2b87a2f854430b57c644abd85.torrent' found in cache: C:\Users\hmy01\AppData\Local\Him188\Ani\cache\torrent-data\anitorrent\torrentFiles\792322585.txt, length=13887
2025-09-12 14:14:46,847 [INFO ] app.torrent.anitorrent.AnitorrentTorrentDownloader: Found existing session
2025-09-12 14:14:46,847 [INFO ] app.torrent.anitorrent.AnitorrentTorrentDownloader: withHandleTaskQueue: executed 0 delayed tasks
2025-09-12 14:14:46,847 [INFO ] app.domain.media.cache.engine.TorrentMediaCacheEngine: local-file-system: waiting for files
2025-09-12 14:14:46,848 [INFO ] app.domain.media.cache.engine.TorrentMediaCacheEngine: local-file-system: Selected file to download: TorrentFileEntryImpl(index=0, length=668507476, relativePath='[DMG&Sakurato][Silent_Witch][09.5][1080P][GB].mp4')
2025-09-12 14:14:46,851 [ERROR] app.ui.subject.cache.SubjectCacheViewModelImpl: Unhandled exception in background scope for viewmodel me.him188.ani.app.ui.subject.cache.SubjectCacheViewModelImpl, coroutineContext: [me.him188.ani.app.ui.foundation.AbstractViewModel$createBackgroundScope$$inlined$CoroutineExceptionHandler$1@519ff553, StandaloneCoroutine{Cancelling}@1b75936b, Dispatchers.Default]
java.lang.IllegalStateException: Sealed class 'me.him188.ani.datasources.api.EpisodeSort.Special' cannot be serialized as base class 'me.him188.ani.datasources.api.EpisodeSort' because it has property name that conflicts with JSON class discriminator 'type'. You can either change class discriminator in JsonConfiguration, rename property with @SerialName annotation or fall back to array polymorphism
	at kotlinx.serialization.json.internal.PolymorphicKt.validateIfSealed(Polymorphic.kt:62)
	at kotlinx.serialization.json.internal.PolymorphicKt.access$validateIfSealed(Polymorphic.kt:1)
	at kotlinx.serialization.json.internal.StreamingJsonEncoder.encodeSerializableValue(StreamingJsonEncoder.kt:252)
	at kotlinx.serialization.encoding.AbstractEncoder.encodeSerializableElement(AbstractEncoder.kt:80)
	at me.him188.ani.datasources.api.MediaCacheMetadata.write$Self$datasource_api(MediaCacheMetadata.kt:28)
	at me.him188.ani.datasources.api.MediaCacheMetadata$$serializer.serialize(MediaCacheMetadata.kt:28)
	at me.him188.ani.datasources.api.MediaCacheMetadata$$serializer.serialize(MediaCacheMetadata.kt:28)
	at kotlinx.serialization.json.internal.StreamingJsonEncoder.encodeSerializableValue(StreamingJsonEncoder.kt:259)
	at kotlinx.serialization.encoding.AbstractEncoder.encodeSerializableElement(AbstractEncoder.kt:80)
	at me.him188.ani.app.domain.media.cache.storage.MediaCacheSave.write$Self$app_data(MediaCacheStorage.kt:118)
	at me.him188.ani.app.domain.media.cache.storage.MediaCacheSave$$serializer.serialize(MediaCacheStorage.kt:118)
	at me.him188.ani.app.domain.media.cache.storage.MediaCacheSave$$serializer.serialize(MediaCacheStorage.kt:118)
	at kotlinx.serialization.json.internal.StreamingJsonEncoder.encodeSerializableValue(StreamingJsonEncoder.kt:259)
	at kotlinx.serialization.encoding.AbstractEncoder.encodeSerializableElement(AbstractEncoder.kt:80)
	at kotlinx.serialization.internal.CollectionLikeSerializer.serialize(CollectionSerializers.kt:69)
	at kotlinx.serialization.json.internal.StreamingJsonEncoder.encodeSerializableValue(StreamingJsonEncoder.kt:259)
	at kotlinx.serialization.json.internal.JsonStreamsKt.encodeByWriter(JsonStreams.kt:99)
	at kotlinx.serialization.json.JvmStreamsKt.encodeToStream(JvmStreams.kt:25)
	at me.him188.ani.app.data.persistent.DataStoreMP_jvmKt$asDataStoreSerializer$1.writeTo(DataStoreMP.jvm.kt:50)
	at androidx.datastore.core.FileWriteScope.writeData(FileStorage.kt:201)
	at androidx.datastore.core.DataStoreImpl$writeData$2.invokeSuspend(DataStoreImpl.kt:353)
	at androidx.datastore.core.DataStoreImpl$writeData$2.invoke(DataStoreImpl.kt)
	at androidx.datastore.core.DataStoreImpl$writeData$2.invoke(DataStoreImpl.kt)
	at androidx.datastore.core.FileStorageConnection.writeScope(FileStorage.kt:118)
	at androidx.datastore.core.DataStoreImpl.writeData$datastore_core(DataStoreImpl.kt:348)
	at androidx.datastore.core.DataStoreImpl$transformAndWrite$2.invokeSuspend(DataStoreImpl.kt:337)
	at androidx.datastore.core.DataStoreImpl$transformAndWrite$2.invoke(DataStoreImpl.kt)
	at androidx.datastore.core.DataStoreImpl$transformAndWrite$2.invoke(DataStoreImpl.kt)
	at androidx.datastore.core.SingleProcessCoordinator.lock(SingleProcessCoordinator.kt:41)
	at androidx.datastore.core.DataStoreImpl.transformAndWrite(DataStoreImpl.kt:329)
	at androidx.datastore.core.DataStoreImpl.handleUpdate(DataStoreImpl.kt:237)
	at androidx.datastore.core.DataStoreImpl.access$handleUpdate(DataStoreImpl.kt:48)
	at androidx.datastore.core.DataStoreImpl$writeActor$3.invokeSuspend(DataStoreImpl.kt:207)
	at androidx.datastore.core.DataStoreImpl$writeActor$3.invoke(DataStoreImpl.kt)
	at androidx.datastore.core.DataStoreImpl$writeActor$3.invoke(DataStoreImpl.kt)
	at androidx.datastore.core.SimpleActor$offer$2.invokeSuspend(SimpleActor.kt:121)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:124)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:89)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:586)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:820)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:717)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:704)
```

--- a/datasource/api/src/commonMain/kotlin/EpisodeSort.kt
+++ b/datasource/api/src/commonMain/kotlin/EpisodeSort.kt
@@ -9,6 +9,7 @@
 
 package me.him188.ani.datasources.api
 
+import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
 import kotlinx.serialization.protobuf.ProtoNumber
 import me.him188.ani.datasources.api.EpisodeSort.Normal
@@ -92,7 +93,7 @@ sealed class EpisodeSort : Comparable<EpisodeSort> {
 
     @Serializable
     class Special internal constructor(
-        @ProtoNumber(1) val type: EpisodeType,
+        @ProtoNumber(1) @SerialName("episodeType") val type: EpisodeType,
         @ProtoNumber(2) override val number: Float?,
     ) : EpisodeSort() {
         override val raw: String get() = "${type.value}${getNumberStr(number)}" // "SP01"

